### PR TITLE
Refactor/use decorator pattern

### PR DESCRIPTION
**Pull request by mike4git**
Ansatz für einen decorated DefaultConverter

@jdreesen Geht das in die Richtung, die Du meintest?!

### CODE REVIEW

File: config/services.yaml [hunk 1]
```diff
@@ -16,4 +16,7 @@ services:
     default.converter.with.cache:
         abstract: true
         class: Neusta\ConverterBundle\Converter\DefaultCachedConverter
+        decorates: default.converter
+        arguments:
+            $inner: '@default.converter'
```
Comment (jdreesen) [inline config/services.yaml]:
```suggestion
            $inner: '@.inner'
```

See: https://symfony.com/doc/5.4/service_container/service_decoration.html

File: tests/app/config/services.yaml [hunk 2]
```diff
@@ -21,9 +21,7 @@ services:
         parent: 'default.converter.with.cache'
         public: true
         arguments:
-            $factory: '@Neusta\ConverterBundle\Tests\Fixtures\Factory\PersonFactory'
-            $populators:
-                - '@Neusta\ConverterBundle\Tests\Fixtures\Populator\PersonNamePopulator'
+            $inner: '@default.converter'
```
Comment (jdreesen) [inline tests/app/config/services.yaml]: Does Symfony force us to re-define the `$inner` argument here? Would be sad...

```suggestion
            $inner: '@.inner'
```

**Review by jdreesen — CHANGES_REQUESTED** [review]
Yes, that's what I meant. Very nice!

**Comment by jdreesen** [comment]
I think this is a BC break, too.

**Comment by mike4git** [comment]
I don't think it's a BC because it only changes implementation details. 
Anyway your new PRs will be so it does not play a role now.

**Comment by jdreesen** [comment]
I'd say it is because it changes how you have to configure the cached converter.

**Pull request merged**